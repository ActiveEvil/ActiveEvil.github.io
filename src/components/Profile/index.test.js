import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Component from './';

describe('Profile component', () => {

  test('does not render when there is no active character profile', () => {
    const wrapper = shallow(
      <Component />
    );

    expect(wrapper.find('.container').exists()).toBe(false);
  });

  test('renders a character profile', () => {
    const wrapper = shallow(
      <Component
        character={{
          birth_year: '44BBY',
          eye_color: 'black',
          films: ['A New Hope'],
          gender: 'male',
          hair_color: 'n/a',
          height: '173',
          homeworld: 'Rodia',
          mass: '74',
          name: 'Greedo',
          skin_color: 'green',
          species: 'Rodian',
        }}
      />
    );

    expect(wrapper.find('.container').exists()).toBe(true);
  });

  test('renders as expected', () => {
    const component = renderer.create(
      <Component
        character={{
          birth_year: '44BBY',
          eye_color: 'black',
          films: ['A New Hope'],
          gender: 'male',
          hair_color: 'n/a',
          height: '173',
          homeworld: 'Rodia',
          mass: '74',
          name: 'Greedo',
          skin_color: 'green',
          species: 'Rodian',
        }}
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
