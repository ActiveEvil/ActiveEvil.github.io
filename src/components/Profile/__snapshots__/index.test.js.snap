// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders as expected 1`] = `
<section
  className="container"
>
  <h2
    className="title"
  >
    Greedo
  </h2>
  <table>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          Homeworld
        </th>
        <td>
          Rodia
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Born
        </th>
        <td>
          44BBY
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Species
        </th>
        <td>
          Rodian
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Gender
        </th>
        <td>
          male
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Height
        </th>
        <td>
          173
          cm
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Weight
        </th>
        <td>
          74
          kg
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Hair colour
        </th>
        <td>
          n/a
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Eye colour
        </th>
        <td>
          black
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          Appears in
        </th>
        <td>
          <span
            className="film"
          >
            A New Hope
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
